packages/peregrine/lib/talons/MegaMenu/useMegaMenu.js

```diff
@@ -119,9 +119,35 @@ export const useMegaMenu = (props = {}) => {
         }
     }, [findActiveCategory, location.pathname, megaMenuData]);
 
+    const useOutsideAlerter = ref => {
+        useEffect(() => {
+            // Reset menu if clicked outside
+            const handleClickOutside = e => {
+                if (ref.current && !ref.current.contains(e.target)) {
+                    props.setSubMenuState(false);
+                    setDisableFocus(true);
+                }
+            };
+
+            // Bind the event listener to both mouse and key events
+            window.addEventListener('mousedown', handleClickOutside);
```

Comment:
We can use ` window` because we want to use add SSR 

https://stephencook.dev/blog/using-window-in-react-ssr/